### PR TITLE
Allow non-gov.uk email addresses to be invited

### PR DESCRIPTION
**Pull request by tommotaylor**
A requirement as often GovWifi admins will be external suppliers eg: bt.com

### CODE REVIEW

File: app/controllers/users/invitations_controller.rb [hunk 1]
```diff
@@ -14,15 +14,25 @@ def add_organisation_to_params
...
+  end
+
+  def email_passes_validation?
+    invite_params[:email].match(URI::MailTo::EMAIL_REGEXP).present?
+  end
+
+  def set_user_object_with_errors
     @user = User.new(invite_params)
-    !@user.validate
+    @user.errors.add(:email, " must be a valid email address")
```
Comment (emileswarts) [inline app/controllers/users/invitations_controller.rb]: Wonder if we can push this down into a model validation.
Validation can have a conditional to check whether the user is being created as an admin or invited.

Comment (tommotaylor) [reply app/controllers/users/invitations_controller.rb]: `!@user.validate` was using the model validation on the user. It checked that the email was present, valid email and that it was on the whitelist.

However in this instance we want to skip the whitelist check, and you can't do conditional validation like `!@user.validate(except: :whitelist)`. So in this instance I'm just checking that the email is an email manually (which covers presence and valid format).

I thought about separating validations from the model using something like [`dry-validation` gem](https://github.com/dry-rb/dry-validation) , but thought its overkill for one email field.

Comment (tommotaylor) [reply app/controllers/users/invitations_controller.rb]: Or were you thinking an instance method on `User` like `@user.email_valid`?

Comment (emileswarts) [reply app/controllers/users/invitations_controller.rb]: Was thinking of modifying the user model like this:

```
  validate :email_on_whitelist, on: :create, unless: :invited?
```

**Pull request merged**